packages/components/table/src/table-column/index.ts

```diff
@@ -150,6 +151,11 @@ export default defineComponent({
           updateColumnOrder
         )
     })
+
+    onUpdated(() => {
+      owner.value.store.updateColumns()
+    })
```

Comment:
IMO, The code here adds extra performance overhead.

Maybe we can consider handling `selection` changes in [L43](https://github.com/element-plus/element-plus/blob/dev/packages/components/table/src/table-column/watcher-helper.ts#L43)